control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
+                {
+                    "name": "components",
+                    "schema": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "compose_global_directory",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "object",
+                    "additionalProperties": true
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "compose_local_directory",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "object",
+                    "additionalProperties": true
+                },
+                "required": true
+            },
+            "description": "Compose a dictionary with the local directory."
+        },
+        {
+            "name": "get_component_names",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "clean_addresses",
+            "params": [
+                {
+                    "name": "expiration_time",
+                    "schema": {
+                        "type": "number"
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            },
+            "description": "Clean all expired addresses from the directory.\n\n        :param float expiration_time: Expiration limit in s.\n        "
+        },
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
+
+#### Actor
+
+An Actor is a {ref}`control_protocol.md#Component`.
+Additionally, it MUST offer the following methods.
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Actor",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "get_parameters",
+            "params": [
+                {
+                    "name": "parameters",
+                    "schema": {
+                        "anyOf": [
+                            {
+                                "type": "array",
```

Comment:
duplicated? something's wrong here I think.

Reply:
In python I defined `parameters: tuple[str] | list[str]` as it does not matter. Both, tuple and list, translate to array....

I fix it.